### PR TITLE
Soften name validation warnings to avoid CI issues

### PR DESCRIPTION
**Pull request by wincent**
A few things here:

- Downgrade `console.error` to `console.warn`.
- Strip "Error: " prefix from messages that can trip up CIs.
- Note in warning message that it will eventually become a hard error.

### CODE REVIEW

**Comment by leebyron** [comment]
Thanks for doing the research and adding tests. This looks great. Feel free to merge.

I think beyond this we should follow-up with with a slightly more invasive change to separate the construction of a schema from the validation of a schema. 

When building a "source of truth" schema (e.g. you run a node.js server, and use GraphQL.js to present a GraphQL service) then we should encourage validation - strict validation - to ensure the schema is well formed. I'm not sure what the best way to accomplish this would be and am open to ideas. Perhaps `execute()` or other operations dependent on a valid schema would check some internally-set flag for "has been validated", and if it's not set, run validation? Just an idea.

When building a "reference" schema (e.g. you used introspection to learn about a schema, and have rebuilt it to power a tool like GraphiQL) then we have very different validation needs. Assumptions made by execution like an object adhering to its implemented interfaces, or (in this case) names being well formed may not break assumptions of the tools being used - we're both paying a computation cost to perform this validation and we're forcing tools to be more strict than they might need to be. Should GraphiQL stop working or report errors if it's used against a server that has an invalid schema? If it should, how do we define that line?

This diff is a great incremental step to easing the pain of this particular change, but it would be awesome to see a broader rethink of how schema validation fits into the whole story.

File: src/utilities/__tests__/assertValidName-test.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/**
```
Comment (asiandrummer) [inline src/utilities/__tests__/assertValidName-test.js]: Love this

File: src/utilities/assertValidName.js [hunk 2]
```diff
@@ -28,12 +29,13 @@ export function assertValidName(
   if (!isIntrospection && name.slice(0, 2) === '__' && !hasWarnedAboutDunder) {
     hasWarnedAboutDunder = true;
     /* eslint-disable no-console */
-    if (console && console.error) {
+    if (console && console.warn) {
       const error = new Error(
         `Name "${name}" must not begin with "__", which is reserved by ` +
-        'GraphQL introspection.'
+        'GraphQL introspection. In a future release of graphql this will ' +
```
Comment (asiandrummer) [inline src/utilities/assertValidName.js]: nit: graphql -> graphql-js?

Comment (wincent) [inline src/utilities/assertValidName.js]: I went with `graphql` because that is the NPM package name. `graphql-js` is only the name of the GitHub repo.

**Comment by wincent** [comment]
Split off https://github.com/graphql/graphql-js/issues/719 to carry out discussion about the deeper changes to the validation model.

**Pull request merged**